lib/private/Collaboration/Collaborators/GroupPlugin.php

```diff
@@ -37,34 +37,26 @@
 use OCP\Share\IShare;
 
 class GroupPlugin implements ISearchPlugin {
-	/** @var bool */
-	protected $shareeEnumeration;
-	/** @var bool */
-	protected $shareWithGroupOnly;
-	/** @var bool */
-	protected $shareeEnumerationInGroupOnly;
-	/** @var bool */
-	protected $groupSharingDisabled;
-
-	/** @var IGroupManager */
-	private $groupManager;
-	/** @var IConfig */
-	private $config;
-	/** @var IUserSession */
-	private $userSession;
-
-	public function __construct(IConfig $config, IGroupManager $groupManager, IUserSession $userSession) {
-		$this->groupManager = $groupManager;
-		$this->config = $config;
-		$this->userSession = $userSession;
+	protected bool $shareeEnumeration;
 
+	protected bool $shareWithGroupOnly;
+
+	protected bool $shareeEnumerationInGroupOnly;
+
+	protected bool $groupSharingDisabled;
+
+	public function __construct(
+		private IConfig $config,
+		private IGroupManager $groupManager,
+		private IUserSession $userSession,
+	) {
 		$this->shareeEnumeration = $this->config->getAppValue('core', 'shareapi_allow_share_dialog_user_enumeration', 'yes') === 'yes';
 		$this->shareWithGroupOnly = $this->config->getAppValue('core', 'shareapi_only_share_with_group_members', 'no') === 'yes';
 		$this->shareeEnumerationInGroupOnly = $this->shareeEnumeration && $this->config->getAppValue('core', 'shareapi_restrict_user_enumeration_to_group', 'no') === 'yes';
 		$this->groupSharingDisabled = $this->config->getAppValue('core', 'shareapi_allow_group_sharing', 'yes') === 'no';
 	}
 
-	public function search($search, $limit, $offset, ISearchResult $searchResult) {
+	public function search($search, $limit, $offset, ISearchResult $searchResult): bool {
```

Comment:
Could the types be known? Then you want to check these

```suggestion
	public function search(string $search, int $limit, int $offset, ISearchResult $searchResult): bool {
```

Reply:
same.